tests/queries/0_stateless/01660_join_or_any.sql

```diff
@@ -0,0 +1,52 @@
+SET joined_subquery_requires_alias = 0;
+SET max_threads = 1;
+
+drop table if exists tab1;
+drop table if exists tab2;
+drop table if exists tab3;
+
+create table tab1 (a1 Int32, b1 Int32) engine = MergeTree order by a1;
+create table tab2 (a2 Int32, b2 Int32) engine = MergeTree order by a2;
+create table tab3 (a3 Int32, b3 Int32) engine = MergeTree order by a3;
+
+insert into tab1 values (1, 2);
+insert into tab2 values (2, 3);
+insert into tab2 values (6, 4);
+insert into tab3 values (2, 3);
+insert into tab3 values (5, 4);
+insert into tab3 values (100, 4);
+
+select 'join on OR chain (any left)';
+select a2, b2 from tab2 any left join tab3 on a2 = a3 or b2 = b3;
+select '==';
+select a3, b3 from tab2 any left join tab3 on a2 = a3 or b2 = b3;
+select '==';
+select a2, b2, a3, b3 from tab2 any left join tab3 on a2 = a3 or b2 = b3;
+select '==';
+select a1 from tab1 any left join tab2 on b1 + 1 = a2 + 1 or a1 + 4 = b2 + 2;
+select '==';
+select a1, b2 from tab1 any left join tab2 on b1 + 1 = a2 + 1 or a1 + 4 = b2 + 2;
+select '==';
+select a1, b1, a2, b2 from tab1 any left join tab2 on b1 + 1 = a2 + 1 or a1 + 4 = b2 + 2;
+select '==';
+select a2, b2 + 1 from tab1 any left join tab2 on b1 + 1 = a2 + 1 or a1 + 4 = b2 + 2;
+
+
+select 'join on OR chain (any right)';
+select a2, b2 from tab2 any right join tab3 on a2 = a3 or b2 = b3;
+select '==';
+select a3, b3 from tab2 any right join tab3 on a2 = a3 or b2 = b3;
```

Comment:
Interesting. Thanks.

Reply:
Trying to write tests I realized that the best I can do is to treat actual behavior as a reference because I don't really understand 'any_join_distinct_right_table_keys = 1' logic.
:)  select tab2.*, tab3.* from tab2 any right join tab3 on  a2 = a3;
─a2┬b2┬a3─┬b3┐
 |   2  │  3 │  2 │  3 │
│  0 │  0 │   5 │  4 │
│  0 │  0 │ 100 │  4 │
:) select tab2.*, tab3.* from tab2 any right join tab3 on b2 = b3;
─a2┬b2┬a3─┬b3┐
│  2 │  3 │  2 │  3 │
│  6 │  4 │  5 │  4 │
Why do we have nulls in the first case and do not have in the second (100 4  does not exist)? @vdimir, could you shed some light on this, please.

Reply:
Yes, with `any_join_distinct_right_table_keys = 1` we have only one row from right table for each unique key value. With `any_join_distinct_right_table_keys = 0` keys in right table can be repeated. So, `any_join_distinct_right_table_keys = 1` is aimed to make `LEFT` and `RIGHT` `ANY JOIN` more consistent.